linta-android/src/main/java/com/swvl/lint/checks/DuplicateResourceFilesDetector.kt

```diff
@@ -85,12 +85,21 @@ class DuplicateResourceFilesDetector : ResourceXmlDetector() {
                 i++
             }
         }
+        // Remove comment nodes.
+        else if (node.nodeType == Element.COMMENT_NODE) {
+            node.parentNode.removeChild(node)
+        }
 
         // Do the same with all children.
-        val childrenCount = node.childNodes.length
-        for (i in 0 until childrenCount) {
+        var i = 0
+        while (i < node.childNodes.length) {
             val child = node.childNodes.item(i)
-            removeToolsNamespaceAttributes(child)
+            removeToolsNamespaceAttributesAndComments(child)
+
+            // Only increase counter if node isn't a comment
+            if (child.nodeType == Element.COMMENT_NODE)
+                continue
```

Comment:
When not adding curly brackets for single-line if statements, it would be better to keep the statement in one line, to not confuse the readers of `i++` that it's a part of the if statement.

```suggestion
            if (child.nodeType == Element.COMMENT_NODE) continue
```

Reply:
Actually I'd prefer to keep it as is, because the indentation is what makes it clear what is part of the if statement and what is not

Reply:
I wouldn't prefer that. It would look error-prone if it's on more than one line.

Check [Section 7.4 in pages 16 and 17 in Oracle's Java Code Conventions](https://www.oracle.com/technetwork/java/codeconventions-150003.pdf). Kotlin follows the same.

Reply:
I'll prefer to add the braces then